lightwood/analysis/analyze.py

```diff
@@ -103,7 +104,13 @@ def model_analyzer(
     # ------------------------- #
     # Additional Analysis Blocks
     # ------------------------- #
-    for block in analysis_blocks:
-        runtime_analyzer = block.compute(runtime_analyzer, **{})
+    if len(analysis_blocks) > 0:
+        exec(lightwood.api.json_ai.IMPORTS_FOR_EXTERNAL_DIRS, globals())
+        exec(lightwood.api.json_ai.IMPORT_EXTERNAL_DIRS, globals())
+
+        for dirpath in analysis_blocks:
+            module, block_name = dirpath.split(".")
+            block = getattr(eval(module), block_name)()
+            runtime_analyzer = block.analyze(runtime_analyzer, **{})
```

Comment:
Ok, I think this is a really bad idea to be honest.

As in, this is basically moving dynamic code generation from json_ai, where it's at least constrained, into the model_analyzer.

If we go the route of "every module can do dynamic importing" (outside of very specific cases, e.g. analyzer can dynamically import sklearn acc function from strings) then the design can get messy.

Why not have the user's specify the blocks in the json_ai and import them at json_ai creation time?

Reply:
Very good point, let's do that instead.